### PR TITLE
Customize route for image-registry on build02

### PR DESCRIPTION
**Pull request by hongkailiu**
/cc @stevekuznetsov @alvaroaleman

### CODE REVIEW

File: clusters/build-clusters/02_cluster/openshift-image-registry/router.md [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+...
+
+$ oc --context build02 get route -n openshift-image-registry
+NAME            HOST/PORT                           PATH   SERVICES         PORT    TERMINATION   WILDCARD
+public-routes   registry.build02.ci.openshift.org          image-registry   <all>   reencrypt     None
+
+$ podman pull registry.build02.ci.openshift.org/ci/applyconfig --tls-verify=false
+```
+
+* Create a secret with your route’s TLS keys:
```
Comment (hongkailiu) [inline clusters/build-clusters/02_cluster/openshift-image-registry/router.md:39]: I have reached to this point.
Will continue after merge: The secret `cert-issuer` is need for cert-manager.

File: clusters/build-clusters/02_cluster/openshift-image-registry/router.md [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+  routes:
+  - hostname: registry.build02.ci.openshift.org
+    name: public-routes
+...
+
+$ oc --context build02 get route -n openshift-image-registry
+NAME            HOST/PORT                           PATH   SERVICES         PORT    TERMINATION   WILDCARD
+public-routes   registry.build02.ci.openshift.org          image-registry   <all>   reencrypt     None
+
+$ podman pull registry.build02.ci.openshift.org/ci/applyconfig --tls-verify=false
```
Comment (hongkailiu) [inline clusters/build-clusters/02_cluster/openshift-image-registry/router.md:36]: The DNS setup is working.
We are on the way of removing `--tls-verify=false`.

**Comment by hongkailiu** [comment]
/hold

Maybe the namespace will be created by the installation of cert-manager.
Let us wait to avoid conflicting.

**Comment by hongkailiu** [comment]
non-operator installation works:

```
echo $site
registry.build02.ci.openshift.org

curl --insecure -v "https://${site}" 2>&1 | awk 'BEGIN { cert=0 } /^\* Server certificate:/ { cert=1 } /^\*/ { if (cert) print }'
* Server certificate:
*  subject: CN=registry.build02.ci.openshift.org
*  start date: May 29 01:06:13 2020 GMT
*  expire date: Aug 27 01:06:13 2020 GMT
*  issuer: C=US; O=Let's Encrypt; CN=Let's Encrypt Authority X3
*  SSL certificate verify ok.
* Connection #0 to host registry.build02.ci.openshift.org left intact
* Closing connection 0
```

**Comment by hongkailiu** [comment]
We need somehow monitor the validity of the certificate: in case the renewal stops working, we notice.

File: clusters/build-clusters/02_cluster/cert-manager/cert-issuer_clusterissuer.yaml [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+kind: ClusterIssuer
+metadata:
+  name: cert-issuer
+spec:
+  acme:
+    email: openshift-ci-robot@redhat.com
+    server: https://acme-v02.api.letsencrypt.org/directory
+    #server: https://acme-staging-v02.api.letsencrypt.org/directory
+    privateKeySecretRef:
+      name: cert-issuer-account-key
```
Comment (stevekuznetsov) [inline clusters/build-clusters/02_cluster/cert-manager/cert-issuer_clusterissuer.yaml:11]: Where does this come from?

Comment (hongkailiu) [reply clusters/build-clusters/02_cluster/cert-manager/cert-issuer_clusterissuer.yaml:11]: I googled it too yesterday.
It is created by cert-manager (we just need to tell a secret name to use).
https://github.com/jetstack/cert-manager/issues/1751

We actually do not need it.
(in the acme.sh, we do not use it)

Comment (stevekuznetsov) [reply clusters/build-clusters/02_cluster/cert-manager/cert-issuer_clusterissuer.yaml:11]: SGTM

**Pull request merged**